File: Ranks.js
class Ranks{

    rankList = ['RCT', 'CDT', 'PVT', 'PV2', 'PFC', 'SPC', 'LCPL', 'CPL', 'SGT', 'SSG', 'LT2', 'LT', 'LCDR', 'CDR', 'CAPT', 'VADM', 'ADM', 'HADM']

    rankAbbrevs = [
        {rank: 'Recruit', abbrev: 'RCT', num: 0 },
        {rank: 'Cadet', abbrev: 'CDT', num: 1  },
        {rank: 'Private', abbrev: 'PVT', num: 2  },
        {rank: 'Private Second Class', abbrev: 'PV2', num: 3 },
        {rank: 'Private First Class', abbrev: 'PFC', num: 4 },
        {rank: 'Specialist', abbrev: 'SPC', num: 5 },
        {rank: 'Lance Corporal', abbrev: 'LCPL', num: 6 },
        {rank: 'Corporal', abbrev: 'CPL', num: 7 },
        {rank: 'Sergeant', abbrev: 'SGT' , num: 8 },
        {rank: 'Staff Sergeant', abbrev: 'SSG' , num: 9 },
        {rank: 'Second Lieutenant', abbrev: 'LT2' , num: 10 },
        {rank: 'Lieutenant', abbrev: 'LT' , num: 11 },
        {rank: 'Lieutenant Commander', abbrev: 'LCDR' , num: 12 },
        {rank: 'Commander', abbrev: 'CDR' , num: 13 },
        {rank: 'Captain', abbrev: 'CPT' , num: 14 },
        {rank: 'Vice Admiral', abbrev: 'VADM' , num: 15 },
        {rank: 'Admiral', abbrev: 'ADM' , num: 16 },
        {rank: 'High Admiral', abbrev: 'HADM' , num: 17 },
    ]

    
    ClearAllRanks(nick){
        for(var i = 0; i < this.rankAbbrevs.length; i++){
        nick = nick.replace(`${this.rankAbbrevs[i].abbrev}. `, '');
        }

        return nick
    }

    RankByNumber(num){
        for(var i = 0; i < this.rankAbbrevs.length; i++){
            if(this.rankAbbrevs[i].num === num){
                return this.rankAbbrevs[i]
            }
        }
    }

    GetRank(rankName){
        for(var i = 0; i < this.rankAbbrevs.length; i++){
          if(rankName.toUpperCase() === this.rankAbbrevs[i].abbrev){
            return this.rankAbbrevs[i]
          }
        }
    
        return null;
      }

    HasRank(nick){
        for(var i = 0; i< this.rankAbbrevs.length; i++){
            let nsplit = nick.split(/ +/);;
            if(nsplit[0] === `${this.rankAbbrevs[i].abbrev}.`){
            return this.rankAbbrevs[i]
            }
        }
        return null;
    }
}

module.exports = Ranks;

File: C#/multicorp-bot/POCO/Ranks.cs
using System.Collections.Generic;
using DSharpPlus.Entities;

namespace multicorp_bot {
    public class Ranks {

        private List<Rank> MilRanks = new List<Rank> () {
            new Rank () { RankName = "Recruit", Abbreviation = "RCT", Number = 0 },
            new Rank () { RankName = "Cadet", Abbreviation = "CDT", Number = 1 },
            new Rank () { RankName = "Private", Abbreviation = "PVT", Number = 2 },
            new Rank () { RankName = "Private Second Class", Abbreviation = "PV2", Number = 3 },
            new Rank () { RankName = "Private First Class", Abbreviation = "PFC", Number = 4 },
            new Rank () { RankName = "Specialist", Abbreviation = "SPC", Number = 5 },
            new Rank () { RankName = "Lance Corporal", Abbreviation = "LCPL", Number = 6 },
            new Rank () { RankName = "Corporal", Abbreviation = "CPL", Number = 7 },
            new Rank () { RankName = "Sergeant", Abbreviation = "SGT", Number = 8 },
            new Rank () { RankName = "Staff Sergeant", Abbreviation = "SSG", Number = 9 },
            new Rank () { RankName = "Second Lieutenant", Abbreviation = "2LT", Number = 10 },
            new Rank () { RankName = "Lieutenant", Abbreviation = "LT", Number = 11 },
            new Rank () { RankName = "Lieutenant Commander", Abbreviation = "LCDR", Number = 12 },
            new Rank () { RankName = "Commander", Abbreviation = "CMDR", Number = 13 },
            new Rank () { RankName = "Captain", Abbreviation = "CPT", Number = 14 },
            new Rank () { RankName = "Vice Admiral", Abbreviation = "VADM", Number = 15 },
            new Rank () { RankName = "Admiral", Abbreviation = "ADM", Number = 16 },
            new Rank () { RankName = "High Admiral", Abbreviation = "HADM", Number = 17 },
        };

        //Update the abbreviations for every member
        public void UpdateAllRanks (DiscordGuild guild) {

        }

        //Promote a member
        public void Promote (DiscordGuild guild) {

        }

        //Demote a member
        public void Demote (DiscordGuild guild) {

        }

        //Assignst the starter rank
        public void Recruit (DiscordGuild guild) {

        }
    }
}